index.d.ts

```diff
@@ -1,271 +1,829 @@
+// Needed for BrowserWebSocket
+/// <reference lib="dom" />
```

Comment:
is it possible to have a index.dom.ts and have that as the dom typings?

Reply:
Well we'd need some sort of postinstall/prepare script which prompts the user if they're using a browser, referencing it won't work as it'll import it as full and basically replicate this. I do have some sort of idea which implements this but it's not that great of a workaround since we still need to maintain 2 sets of typings. Caalyx actually suggested to just copy the typings directly from lib.dom.d.ts instead of referencing it so that we wouldn't get global pollution. I'll have a look at this later though

Reply:
Ended up deciding to just ts-ignore the references instead since the types will be referenced if they are using TypeScript in a browser environment